server/utils/courseParticipationUtils.test.ts

```diff
@@ -354,68 +356,145 @@ describe('CourseParticipationUtils', () => {
       })
     })
 
-    describe('when rows are missing required data', () => {
-      it.each([
-        ['Setting', 'setting', undefined],
-        ['Outcome', 'outcome', undefined],
-        ['Outcome', 'outcome', { status: undefined }],
-        ['Additional detail', 'outcome', { detail: undefined }],
-        ['Source of information', 'source', undefined],
-      ])('omits the %s row when %s is %s', (keyText: GovukFrontendSummaryListRowKey['text'], field: string, value) => {
-        const withoutField = { ...courseParticipationWithName, [field]: value }
+    describe('rows with optional values', () => {
+      describe('setting', () => {
+        describe('when the setting has no location', () => {
+          const withoutSettingLocation: CourseParticipationWithName = {
+            ...courseParticipationWithName,
+            setting: {
+              location: undefined,
+              type: 'community',
+            },
+          }
+
+          it("only displays the setting's type", () => {
+            const { rows } = CourseParticipationUtils.summaryListOptions(withoutSettingLocation, referralId)
+            expect(getRowValueText(rows, 'Setting')).toEqual('Community')
+          })
+        })
 
-        const { rows } = CourseParticipationUtils.summaryListOptions(withoutField, referralId)
-        const fieldRow = getRow(rows, keyText)
+        describe('when there is no setting', () => {
+          const withoutSetting: CourseParticipationWithName = {
+            ...courseParticipationWithName,
+            setting: {},
+          }
 
-        expect(fieldRow).toBeUndefined()
+          it('displays "Not known"', () => {
+            const { rows } = CourseParticipationUtils.summaryListOptions(withoutSetting, referralId)
+            expect(getRowValueText(rows, 'Setting')).toEqual('Not known')
+          })
+        })
       })
 
-      it('only shows the location in the Setting row when setting location is undefined', () => {
-        const withoutSettingType = {
-          ...courseParticipationWithName,
-          setting: { location: 'Stockport', type: undefined },
-        }
+      describe('outcome', () => {
+        describe('when the outcome is incomplete', () => {
+          describe('and there is a yearStarted value', () => {
+            const withOutcomeYearStarted = {
+              ...courseParticipationWithName,
+              outcome: { status: 'incomplete' as CourseParticipationOutcome['status'], yearStarted: 2019 },
```

Comment:
If you wanted to avoid casting here and below you could type `withOutcomeYearStarted` as `CourseParticipationWithName` instead